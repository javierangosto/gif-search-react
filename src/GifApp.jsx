import { useState } from "react";
import { AddCategory, GifGrid } from './components'

export const GifApp = () => {

    const [categories, setCategories] = useState([ 'SMITE' ]);

    const onAddCategory = ( newCategory ) => {

        if ( categories.includes( newCategory ) ) return; //Evita insertar duplicados

        //Ambas formas son correctas
        setCategories( [newCategory, ...categories] );
        //setCategories ( cat => [...categories, value] );
    }

    return (
        <>
            <h1>GifApp</h1>

            <AddCategory 
                onNewCategory = { ( value ) => onAddCategory( value ) }
            />
            
            {
                categories.map ( category =>
                    <GifGrid 
                        key={ category } 
                        category={ category }
                    />
                )
            }
        </>
    )
}
