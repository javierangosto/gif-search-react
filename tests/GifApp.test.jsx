import { render, screen } from "@testing-library/react"
import { GifApp } from "../src/GifApp"


describe('Pruebas en GifApp', () => {
    /*

    */
    test('Debe de hacer match con el snapshot', () => {

        const { container } = render( <GifApp /> );

        expect ( container ).toMatchSnapshot();        

    })
})