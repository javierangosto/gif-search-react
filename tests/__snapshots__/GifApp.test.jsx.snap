// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar GIFs"
      type="text"
      value=""
    />
  </form>
  <h3>
    SMITE
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
